### ISSUE TITLE
Text Field clear button `aria-label` is not localized/customizable

### ISSUE THREAD

**Issue by myabc** [primer/view_components#3804]
## To reproduce:

Render a `Primer::Alpha::TextField` component or the Forms DSL + 'text_field` input.

## Actual behavior

It is not possible to specify an `aria-label`. The `aria-label` is hard-coded (in `app/lib/primer/forms/text_field.html.erb`) to the English word "clear".

## Expected behavior

It should be possible to localize the `aria-label`.

## Related

Relates to #3803

**Comment by siddharthkp** [comment]
Hi @myabc, thanks for taking the time to create the issue.

Things are a bit slow at work right now because of the holidays, I'll try to get you a response next week once folks are back at work :)
